Havtorn/Source/Engine/Graphics/Shaders/Includes/DeferredSamplingFunctions.hlsli

```diff
@@ -180,6 +188,64 @@ float PixelShader_SSAO(float2 uv)
     return ssao;
 }
 
+float SampleMaterialTexture(const int textureIndex, const int textureChannelIndex, const float2 uv)
+{
+    // NR: Texture2D materialChannelTextures[11] is not actually an array of Texture2Ds. In DirectX11,
+    // this is just provided as syntactic sugar, and actually defines a bunch of textures in a row without
+    // a wrapping struct. Because of this, it needs to resolve at compile time which texture to actually
+    // sample from, and so the argument to the "array" must be a literal expression. Hence the switch below.
+    
+    float returnValue = 0.0f;
+    
+    switch (textureIndex)
+    {
+        case 0:
+            returnValue = materialChannelTextures[0].Sample(defaultSampler, uv)[textureChannelIndex];
```

Comment:
Suggestion: is it possible to replace with a _#define name_of_texture_on_slot ?

Reply:
If I understand it correctly, these already exist but are not in use? Found corresponding indices with defines in DeferredShaderStructs.hlsli

Reply:
They are different conceptually and should not be used here in order to avoid confusion, but can add another set of defines